### PR TITLE
[CLEANUP] Remove legacy controller proxy behavior

### PR DESCRIPTION
**Pull request by thoov**
Apart of: #15876

### CODE REVIEW

**Review by rwjblue — APPROVED** [review]
Awesome, thank you!

File: packages/ember-runtime/lib/mixins/-proxy.js [hunk 1]
```diff
@@ -105,11 +105,6 @@ export default Mixin.create({
   unknownProperty(key) {
     let content = get(this, 'content');
     if (content) {
-      deprecate(
-        `You attempted to access \`${key}\` from \`${this}\`, but object proxying is deprecated. Please use \`model.${key}\` instead.`,
-        !this.isController,
-        { id: 'ember-runtime.controller-proxy', until: '3.0.0' }
-      );
       return get(content, key);
     }
```
Comment (Turbo87) [inline packages/ember-runtime/lib/mixins/-proxy.js]: @rwjblue shouldn't we remove the remaining code too instead of removing just the deprecation warning?

Comment (thoov) [reply packages/ember-runtime/lib/mixins/-proxy.js]: @Turbo87 What do you mean by remaining code?

Comment (Turbo87) [reply packages/ember-runtime/lib/mixins/-proxy.js]: the `return get(content, key);` 🤔

Comment (rwjblue) [reply packages/ember-runtime/lib/mixins/-proxy.js]: No, that code supports proxies generally. The deprecation was _only_ for when a Ember.Controller base class mixed in the proxy mixin (which is how Ember.ObjectController and Ember.ArrayController worked). Those two classes have been removed, and this mixin is private so there was just no need for the deprecation for when included in controllers...

Comment (Turbo87) [reply packages/ember-runtime/lib/mixins/-proxy.js]: I see, thanks :)

**Pull request merged**